packages/synthetix-main/contracts/modules/core/AccountModule.sol

```diff
@@ -28,6 +29,22 @@ contract AccountModule is IAccountModule, OwnableMixin, AccountRBACMixin, Associ
         return _getNft(_ACCOUNT_SYSTEM);
     }
 
+    function getAccountPermissions(uint accountId) external view returns (AccountPermission[] memory) {
+        AccountRBAC storage accountRbac = _accountModuleStore().accountsRBAC[accountId];
+
+        uint allPermissionsLength = accountRbac.permissionAddresses.length();
+        AccountPermission[] memory allPermissions = new AccountPermission[](allPermissionsLength);
+        for (uint i = 0; i < allPermissionsLength; i++) {
```

Comment:
For set iterations, I recommend starting `i` at `1`, and not doing the `i + 1` thing when accessing the position.